src/proposal.mligo

```diff
@@ -223,16 +214,11 @@ let unfreeze_proposer_and_voter_token
 let is_level_reached (proposal, target : proposal * blocks): bool =
   Tezos.level > proposal.start_level.blocks + target.blocks
 
-[@inline]
-let frozen_total_supply(store : storage): nat =
-  match Map.find_opt store.frozen_token_id store.total_supply with
-  | Some v -> v
-  | None -> ((failwith "BAD_STATE") : nat)
 
 [@inline]
-let do_total_vote_meet_quorum_threshold (proposal, store : proposal * storage): bool =
+let do_total_vote_meet_quorum_threshold (proposal, config: proposal * config): bool =
   let votes_placed = proposal.upvotes + proposal.downvotes in
-  let total_supply = frozen_total_supply(store) in
+  let total_supply = config.governance_total_supply in
```

Comment:
@pasqu4le Took me a while to update the tests, since there are a few strange cases coming up. But they should be all fine now.

One note is that, since we remove `total_supply`, I have to used `governance_total_supply` here. Not sure if this is to be expected.

This PR is pretty much ready for review, only the specification is left. I'll work on that next.

Reply:
@rinn7e I am a bit doubtful of this change that uses `governance_total_supply` for `frozen_total_supply`.

AFAIU, the governance total supply is the meant to be the total supply of the unfrozen tokens, some of which can be converted to frozen tokens. And the amount of frozen tokens for an account is considered as a measure of how much a voter wants to invest in the DAO.

Also, it does not make sense to use the governace_total_supply as a denominator for calculating the quorum, because AFAIU the denominator should be the maximum possible votes, which is related to the number of frozen tokens, aka, `frozen_total_supply`, and not related to the governance total supply.

@pasqu4le should probably  confirm.

Reply:
I see, if this is the case, then we should not remove `total_supply`. Let me know if we should put this back in.

Reply:
Yeah, @sras is correct, we cannot use the `governance_total_supply` for that we need to store the `total_supply` indeed.

However keep in mind that we only need to store that for frozen tokens, so it can be a single `nat` in the storage, I suggest naming it `frozen_total_supply`.

Reply:
I have added `frozen_total_supply` back now.